### PR TITLE
datagrid category value must be object, it was being set to integer

### PR DESCRIPTION
**Pull request by ahmetakbn**
| Question       | Answer
|----------------|-------------------------------------------------------------
| Bundle version | sonata-project/media-bundle              dev-master b676113 
| Symfony version| symfony/symfony                          v2.8.4
| php version    | PHP 7.0.4-7ubuntu2

# Error
Deleting media object from list view was not being successful although getting the success message. 

# Steps to reproduce
On Media list view press filter once and then try to delete and object. Object is not being deleted. If you press the filter again and again, then you can delete the object.

# Change
Changed the value of category in datagrid to category object. It was being set only category id. If the category id is set by request from tree list, then we find one object by id and context and set the datagrid category to this object.

### CODE REVIEW

**Comment by OskarStark** [comment]
Ping @gregoire

**Comment by greg0ire** [comment]
@OskarStark `greg0ire` != `gregoire`

**Comment by OskarStark** [comment]
This is done by a colleague of mine in a project which i am maintaining.
I told him to create a PR, the change is working, but to be honest i don't know how to test this :(

File: Controller/MediaAdminController.php [hunk 1]
```diff
@@ -80,19 +80,19 @@ public function listAction(Request $request = null)
         $category = $this->container->get('sonata.classification.manager.category')->getRootCategory($context);
 
         if (!$filters) {
-            $datagrid->setValue('category', null, $category->getId());
+            $datagrid->setValue('category', null, $category);
         }
 
         if ($request->get('category')) {
-            $contextInCategory = $this->container->get('sonata.classification.manager.category')->findBy(array(
+            $categoryByContext = $this->container->get('sonata.classification.manager.category')->findOneBy(array(
```
Comment (greg0ire) [inline Controller/MediaAdminController.php]: Please make separate commits for things you find in the way that do not relate to the PR (use `git add --patch`)

Comment (OskarStark) [inline Controller/MediaAdminController.php]: ping @ahmetakbn

**Comment by greg0ire** [comment]
>  I told him to create a PR, the change is working, but to be honest i don't know how to test this :-(

Things are easier to test when they are not in a controller :P I don't see how to easily test this either.

**Comment by ahmetakbn** [comment]
I think now all the changes are related to this PR @greg0ire

**Comment by OskarStark** [comment]
I merged this PR as it is.

* Travis seems not to work currently with our bundle
* The changelog is still the old one
